model/project_ref.go

```diff
@@ -133,6 +134,71 @@ type ProjectRef struct {
 	// Filter/view settings
 	ProjectHealthView ProjectHealthView `bson:"project_health_view" json:"project_health_view" yaml:"project_health_view"`
 	ParsleyFilters    []parsley.Filter  `bson:"parsley_filters,omitempty" json:"parsley_filters,omitempty"`
+
+	// GitHubDynamicTokenPermissionGroups is a list of permission groups for GitHub dynamic access tokens.
+	GitHubDynamicTokenPermissionGroups GitHubDynamicTokenPermissionGroups `bson:"github_dynamic_token_permission_groups,omitempty" json:"github_dynamic_token_permission_groups,omitempty" yaml:"github_dynamic_token_permission_groups,omitempty"`
```

Comment:
I wonder if it makes sense for this to be a map[string]github.InstallationPermissions so that we don't have to iterate through a list when getting it for the requester. This does make the code easier to read so I'm open to either.

Reply:
True. I'm thinking more of the case where we want to get the permission group associated with "gitter_request", so we get` "gitter_request": "secrets-and-other" `, and then if they have created 30 different permission groups, we'll have to iterate over that to get `secrets-and-other`. 
I don't expect these to get too long so optimizing for readability is a good idea. Resolving this.